src/doc/trpl/choosing-guarantees.md

```diff
@@ -0,0 +1,417 @@
+[borrowing][borrowing] in Rust.
+
+[ownership]: ownership.html
+[borrowing]: references-and-borrowing.html
+
+
+
+# Basic pointer types
+
+## `Box<T>`
+
+[`Box<T>`][box] is an "owned pointer" or a "box". While it can hand out borrowed references to the
+data, it is the only owner of the data. In particular, when something like the following occurs:
+
+```rust
+let x = Box::new(1);
+let y = x;
+// x no longer accessible here
+```
+
+Here, the box was _moved_ into `y`. As `x` no longer owns it, the compiler will no longer allow the
+programmer to use `x` after this. A box can similarly be moved _out_ of a function by returning, and
+when a box (one which hasn't been moved) goes out of scope, destructors are run, deallocating the
+inner data.
+
+This abstraction is a low cost abstraction for dynamic allocation. If you want to allocate some
+memory on the heap and safely pass a pointer to that memory around, this is ideal. Note that you
+will only be allowed to share borrowed references to this by the regular borrowing rules, checked at
+compile time.
+
+[box]: ../std/boxed/struct.Box.html
+
+#### Interlude: `Copy`
+
+Move/ownership semantics are not special to `Box<T>`; it is a feature of all types which are not
+`Copy`.
+
+A `Copy` type is one where all the data it logically encompasses (usually, owns) is part of its
+stack representation[^2]. Most types containing pointers to other data are not `Copy`, since there
+is additional data elsewhere, and simply copying the stack representation may accidentally share
+ownership of that data in an unsafe manner.
+
+Types like `Vec<T>` and `String` which also have data on the heap are also not `Copy`. Types
+like the integer/boolean types are `Copy`
+
+`&T` and raw pointers _are_ `Copy`. Even though they do point to further data, they do not "own"
+that data. Whereas `Box<T>` can be thought of as "some data which happens to be dynamically
+allocated", `&T` is thought of as "a borrowing reference to some data". Even though both are
+pointers, only the first is considered to be "data". Hence, a copy of the first should involve a
+copy of the data (which is not part of its stack representation), but a copy of the second only
+needs a copy of the reference. `&mut T` is not `Copy` because mutable aliases cannot be shared, and
+`&mut T` "owns" the data it points to somewhat since it can mutate.
+
+Practically speaking, a type can be `Copy` if a copy of its stack representation doesn't violate
+memory safety.
+
+[^2]: By "stack representation" I mean the data on the stack when a value of this type is held on the stack. For example, a `Vec<T>` has a stack representation of a pointer and two integers (length, capacity). While there is more data behind the indirection of the pointer, it is not part of the stack-held portion of the `Vec`. Looking at this a different way, a type is `Copy` if a `memcopy` of the data copies all the data owned by it.
+
+## `&T` and `&mut T` 
```

Comment:
They provide a zero-cost guarantee, and this blog post is mostly about guarantees -- just that I don't know what to name them.

Also, they _do_ wrap some data, just not by owning, and have a syntax what doesn't use angle brackets. "wrapper types" is not really precisely defined, here I define it to include these :) After all, `Reference<'a, T>` and `MutReference<'a, T>` would be considered to be wrapper types I guess.